src/VisualStudio/Core/Def/Implementation/ProjectSystem/SolutionChangeAccumulator.cs

```diff
@@ -19,19 +22,21 @@ internal class SolutionChangeAccumulator
         /// <see cref="WorkspaceChangeKind.SolutionChanged"/> if we can't give a more precise type.
         /// </summary>
         private WorkspaceChangeKind? _workspaceChangeKind;
+        private readonly List<DocumentId> _documentIdsRemoved = new List<DocumentId>();
 
         public SolutionChangeAccumulator(Solution startingSolution)
         {
             Solution = startingSolution;
         }
 
         public Solution Solution { get; private set; }
+        public IEnumerable<DocumentId> DocumentIdsRemoved => _documentIdsRemoved;
 
         public bool HasChange => _workspaceChangeKind.HasValue;
-        public WorkspaceChangeKind WorkspaceChangeKind => _workspaceChangeKind.Value;
+        public WorkspaceChangeKind WorkspaceChangeKind => _workspaceChangeKind!.Value;
```

Comment:
The compiler also does flow analysis on `Nullable<T>`, so it's flagging this since this function isn't checking HasValue first. Unfortunately there's no way I can annotate this to say that the caller had to use HasChange first.

Reply:
> Unfortunately there's no way I can annotate this to say that the caller had to use HasChange first.

You can `RoslynDebug.Assert` on it.

Reply:
I should clarify: what I want is a way to say that WorkspaceChangeKind can essentially be treated as if it was inside `if (HasChange)`, i.e. inside `if (_workspaceChangeKind.HasValue)` but there's not cross-procedural analysis like that. That way there's not a restatement of the same logic inside HasChange.